elyra/metadata/metadata_app.py

```diff
@@ -376,6 +378,76 @@ def start(self):
+
+    valid_only_flag = Flag("--valid-only", name='valid-only',
+                           description='Only export valid instances (default includes invalid instances)',
+                           default_value=False)
+
+    clean_flag = Flag("--clean", name='clean',
+                      description='Clear out contents of the export directory',
+                      default_value=False)
+
+    directory_option = CliOption("--directory", name='directory',
+                                 description='The local file system path where the exported metadata will be stored',
+                                 required=True)
+
+    # 'Export' flags
+    options: List[Option] = [schema_name_option, valid_only_flag, clean_flag, directory_option]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.metadata_manager = MetadataManager(schemaspace=self.schemaspace)
+
+    def start(self):
+        super().start()  # process options
+
+        include_invalid = not self.valid_only_flag.value
+        schema_name = self.schema_name_option.value
+        directory = self.directory_option.value
+        clean = self.clean_flag.value
+
+        try:
+            if(self.schema_name_option is not None):
+                metadata_instances = self.metadata_manager.get_all(include_invalid=include_invalid,
+                                                                   of_schema=schema_name)
+            else:
+                metadata_instances = self.metadata_manager.get_all(include_invalid=include_invalid)
+        except MetadataNotFoundError:
+            metadata_instances = None
+
+        if not metadata_instances:
+            print("No metadata instances found for schemaspace {}".format(self.schemaspace) +
+                  (" and schema {}".format(schema_name) if schema_name else ""))
+            print("Nothing exported to {}".format(directory))
+            return
+
+        dest_directory = os.path.join(directory, self.schemaspace)
+
+        if not os.path.exists(dest_directory):
+            print("Creating directory structure for {}".format(dest_directory))
+            os.makedirs(dest_directory)
+
+        if clean:
+            print("Cleaning out all files in {}".format(dest_directory))
+            files = [os.path.join(dest_directory, f) for f in os.listdir(dest_directory)]
+            [os.remove(f) for f in files if os.path.isfile(f)]
+
+        print("Exporting metadata instances for schemaspace {}".format(self.schemaspace) +
+              (" and schema {}".format(schema_name) if schema_name else "") +
+              " to {}".format(dest_directory))
+        for metadata in metadata_instances:
+            dict_metadata = metadata.to_dict()
+            output_file = os.path.join(dest_directory, f'{dict_metadata["name"]}.json')
+            with open(output_file, mode='w') as output_file:
+                json.dump(dict_metadata, output_file, indent=4)
```

Comment:
Individual logging - so we know how many and which ones were exported - alright I can make that change. 👍

Reply:
Added code for individual logging of exported metadata (as in case of `list` subcommand). Added an additional column for the export location.

Reply:
@ptitzler - we're contradicting each other (sorry @sandhya-nayak!):
> Not sure if we want to list each instance (this can give users a "warm fuzzy") but I'm okay not for now. One thing we could do in that case is add an indication that the instance is invalid (like is done in list). Hmm, since the inclusion of invalid instances is the default, individual logging might be warranted.

I'm okay only producing a count, although the invalid instances are where things get funky for me.  Perhaps a counter like:

`5 instances exported (2 of which are invalid)`

Reply:
> ... though the invalid instances are where things get funky for me

How so? What would a user do with this extra information?

Reply:
For starters, they would know their export contains invalid instances and which ones are invalid.

Seems like `import` would need an `ignore-invalid` option.

Btw, I've always felt like `--valid-only` was an odd option and even odder default (that invalid instances are included by default).  I think it was because we wanted to have the existence of invalid instances more well-known to the user (thus the default).  But that's a different topic.

Reply:
Ah, I see, so the default behavior. Well, that could be changed in 4.0 (breaking change). I wasn't sure wether the message meant to suggest to the user that some action needs to be taken (e.g. use the `--valid-only` option to ignore the invalid ones, which I agree should not be imported by default.)